src/internet_identity_interface/src/internet_identity/authn_method.rs

```diff
@@ -0,0 +1,39 @@
+use crate::internet_identity::types::{
+    AuthnMethod, AuthnMethodData, AuthnMethodProtection, PublicKeyAuthn, Purpose, WebAuthn,
+};
+use candid::Principal;
+use serde_bytes::ByteBuf;
+
+impl AuthnMethodData {
+    pub fn principal(&self) -> Principal {
```

Comment:
Maybe add a comment describing the meaning of this `principal`?  It doesn't seem immediately obvious.